examples/rcn.py

```diff
@@ -0,0 +1,392 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.1
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# %%
+import os
+import time
+from typing import Dict
+
+import jax
+import matplotlib.pyplot as plt
+
+# %%
+# %matplotlib inline
+import numpy as np
+from jax import numpy as jnp
+from jax import tree_util
+from scipy.ndimage import maximum_filter
+from scipy.signal import fftconvolve
+from sklearn.datasets import fetch_openml
+
+from pgmax.fg import graph, groups
+
+os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
+
+# %% [markdown]
+# # 1. Load the data
+# %%
+hps, vps = 12, 12
+train_size = 20
+test_size = 20
+
+
+def fetch_mnist_dataset(train_size: int, test_size: int, seed: int = 5):
+    """Returns training and test images sampled randomly from the set of MNIST images.
+    Args:
+        train_size: Desired number of training images.
+        test_size: Desired number of test images.
+    Returns:
+        train_set: A list of length train_size containing images from the MNIST train dataset.
+        train_labels: Corresponding labels for the train images.
+        test_set: A list of length test_size containing images from the MNIST test dataset.
+        test_labels: Corresponding labels for the test images.
+    """
+
+    mnist_train_size = 60000
+    dataset = fetch_openml("mnist_784", as_frame=False, cache=True)
```

Comment:
Would be great if there is a way to get a progress bar here. If that's not possible, maybe also print another message reading `Fetching the data` before this line? Otherwise it's just silent and feels strange....

Reply:
Added.